extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -72,24 +71,47 @@ public class DefaultTenantConfigResolver {
     boolean enableHttpForwardedPrefix;
 
     DefaultTenantConfigResolver(BlockingSecurityExecutor blockingExecutor, BeanManager beanManager,
+            Instance<TenantResolver> tenantResolver,
             @ConfigProperty(name = "quarkus.security.events.enabled") boolean securityEventsEnabled,
+            @ConfigProperty(name = "quarkus.oidc.select-tenants-with-issuer") boolean selectTenantsWithIssuer,
             @ConfigProperty(name = "quarkus.http.root-path") String rootPath, TenantConfigBean tenantConfigBean) {
         this.blockingRequestContext = new BlockingTaskRunner<OidcTenantConfig>(blockingExecutor);
         this.securityEventObserved = SecurityEventHelper.isEventObserved(new SecurityEvent(null, (SecurityIdentity) null),
                 beanManager, securityEventsEnabled);
         this.tenantConfigBean = tenantConfigBean;
-        this.pathMatchingTenantResolver = PathMatchingTenantResolver.of(tenantConfigBean.getStaticTenantsConfig(), rootPath,
+
+        var resolvers = new ArrayList<TenantResolver>();
+        if (Boolean.getBoolean(OidcUtils.ANNOTATION_BASED_TENANT_RESOLUTION_ENABLED)) {
+            resolvers.add(createAnnotationBasedTenantResolver());
+        }
+        if (tenantResolver.isResolvable()) {
+            if (tenantResolver.isAmbiguous()) {
+                throw new IllegalStateException("Multiple " + TenantResolver.class + " beans registered");
+            } else {
+                resolvers.add(tenantResolver.get());
+            }
+        }
+        var pathMatchingTenantResolver = PathMatchingTenantResolver.of(tenantConfigBean.getStaticTenantsConfig(), rootPath,
                 tenantConfigBean.getDefaultTenant());
+        if (pathMatchingTenantResolver != null) {
+            resolvers.add(pathMatchingTenantResolver);
+        }
+        if (!tenantConfigBean.getStaticTenantsConfig().isEmpty()) {
+            resolvers.add(new DefaultStaticTenantResolver());
+        }
+        resolvers.add(getTenantIdFromRoutingContext());
+        if (selectTenantsWithIssuer) {
+            IssuerBasedTenantResolver.addIssuerBasedTenantResolver(resolvers, tenantConfigBean.getStaticTenantsConfig(),
+                    tenantConfigBean.getDefaultTenant());
+        }
+        this.staticTenantResolvers = resolvers.toArray(new TenantResolver[0]);
     }
 
     @PostConstruct
     public void verifyResolvers() {
         if (tenantConfigResolver.isResolvable() && tenantConfigResolver.isAmbiguous()) {
             throw new IllegalStateException("Multiple " + TenantConfigResolver.class + " beans registered");
         }
-        if (tenantResolver.isResolvable() && tenantResolver.isAmbiguous()) {
```

Comment:
This one block more than custom `TenantResolver`

Reply:
sorry, I don't get it. Who else than custom `TenantResolver` provides `TenantResolver` bean? Also I didn't change this code, just moved it because it is application scoped bean that can be safely created during the bean initialization.

Reply:
and I need it during the bean creating, that is why I moved it.

Reply:
@michalvavrik I see now you have moved it